File: camerakit/src/main/java/me/kazaf/camerakit/core/ICameraPresenter.java
package me.kazaf.camerakit.core;

import me.kazaf.camerakit.config.ICameraConfig;

/**
 * Created by kazaf on 2018/1/4.
 */

public interface ICameraPresenter {

    void setUserActionCallback(UserActionCallback view);

    void setOnPictureCapturedCallback(OnPictureCapturedCallback pictureCapturedCallback);

    void startPreview();

    void stopPreview();

    void releaseCamera();

    ICameraConfig getConfig();


}

File: camerakit/src/main/java/com/carterchen247/camerakit/core/ICameraPresenter.kt
package com.carterchen247.camerakit.core

interface ICameraPresenter {
    fun setUserActionCallback(view: UserActionCallback)
    fun setOnPictureCapturedCallback(pictureCapturedCallback: OnPictureCapturedCallback)
    fun startPreview()
    fun stopPreview()
    fun releaseCamera()
}